larpandoracontent/LArTrackShowerId/TrackShowerIdFeatureTool.cc

```diff
@@ -25,68 +27,61 @@ using namespace pandora;
 namespace lar_content
 {
 
-ShowerFitFeatureTool::ShowerFitFeatureTool() :
-    m_slidingShowerFitWindow(3)
+TwoDShowerFitFeatureTool::TwoDShowerFitFeatureTool() :
+    m_slidingShowerFitWindow(3),
+    m_slidingLinearFitWindow(10000)
 {
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-void ShowerFitFeatureTool::Run(SupportVectorMachine::DoubleVector &featureVector, const Algorithm *const pAlgorithm,
+void TwoDShowerFitFeatureTool::Run(SupportVectorMachine::DoubleVector &featureVector, const Algorithm *const pAlgorithm,
     const pandora::Cluster *const pCluster)
 {
     if (PandoraContentApi::GetSettings(*pAlgorithm)->ShouldDisplayAlgorithmInfo())
         std::cout << "----> Running Algorithm Tool: " << this->GetInstanceName() << ", " << this->GetType() << std::endl;
 
-    featureVector.push_back(CutClusterCharacterisationAlgorithm::GetShowerFitWidth(pAlgorithm, pCluster, m_slidingShowerFitWindow));
+    float ratio(-1.f);
+    try
+    {
+        const TwoDSlidingFitResult slidingFitResultLarge(pCluster, m_slidingLinearFitWindow, LArGeometryHelper::GetWireZPitch(this->GetPandora()));
+        const float straightLineLength = (slidingFitResultLarge.GetGlobalMaxLayerPosition() - slidingFitResultLarge.GetGlobalMinLayerPosition()).GetMagnitude();
+        if (straightLineLength > std::numeric_limits<double>::epsilon())
```

Comment:
`straightLineLength` is a single float, so probably should compare to `std::numeric_limits<float>::epsilon()`

Most of the other comments below are highlighting this same point :)